expan/core/early_stopping.py

```diff
@@ -159,6 +165,8 @@ def get_or_compile_stan_model(model_file, distribution):
             pickle.dump(sm, f)
     return sm
 
+cacheSamplingResults = False
+samplingResults = {} # memoized sampling results
```

Comment:
yeah that sounds good.
I am fine with all, we just need to make a guideline and stick to it, so that the code will become more consistent and thus more readable.